provider/testdata/Cargo.toml

```diff
@@ -140,7 +140,7 @@ cldr_json_glob = [
 ]
 
 # Git tag or sha1 for the CLDR data used to generate the testdata.
-cldr_json_gitref = "41.0.0"
+cldr_json_gitref = "42.0.0"
```

Comment:
Issue: if this tag changes, you need to re-run `cargo make testdata-download`

You are very good at hitting the shrinking set of invariants that we don't check on CI :) We don't cover this on CI because we don't want to hit the servers to download the data and risk hitting rate limits.

Reply:
Okay, the command I ran is `cargo make testdata-download-sources`.

What do you think about creating a daily or weekly CI job to run this on the latest in `main`?  More of as an automated reminder in case we ever forget.

And FWIW, I ran this command and it didn't cause any errors:
`cargo run -p icu_datagen --features bin --release -- --all-keys --all-locales --overwrite --pretty --cldr-tag 42.0.0 --format dir --icuexport-tag release-72-1 --out /tmp/icu4x-datagen --syntax json`

Reply:
This is great news! It means that datagen and ICU4X can most likely work seamlessly across both CLDR 41 and CLDR 42 without additional changes required.

Reply:
> What do you think about creating a daily or weekly CI job to run this on the latest in main? More of as an automated reminder in case we ever forget.

https://github.com/unicode-org/icu4x/issues/2880